_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,335 @@
+---
+layout: post
+title: "Hacking Strimzi for Cruise Control UI"
+date: 2022-10-31
+author: kyle_liberti
+---
+
+For some time now, Strimzi has used [Cruise Control](https://github.com/linkedin/cruise-control) for automated partition rebalancing and has offered Strimzi custom resources for safely configuring, deploying, and communicating with Cruise Control through the Strimzi Operator.
+These custom resources provide great support for declaritive rebalancing.
+That being said, rather than working with custom resources, you might prefer to interact with applications like Cruise Control through a graphical user interface.
+LinkedIn provides such an interface with their [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application which supports several Cruise Control operations like viewing Kafka cluster status and executing partition rebalances, all with just a couple button clicks.
+Although Strimzi does not offer direct support for the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) it can still be run with and connected to a Strimzi-managed Cruise Control instance.
+
+This post will walk through how to do so! 
+
+**Warning**
+
+Setting up the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) as described in this blog post requires disabling some of the Cruise Control REST API security settings.
+Having these security settings disabled exposes the Cruise Control API to potentially destructive Cruise Control operations that are not coordinated or supported by Strimzi Operator.
+For example, it would allow users to manually decomission brokers and trigger partition rebalances, operations which could interfere with Strimzi operations. 
+If using the Cruise Control UI with a Strimzi-managed Kafka cluster, it is best to stick with using it solely for monitoring your cluster, not for operating on it.
```

Comment:
Is there a way to enforce this read-only mode?

Reply:
Not currently, but there will be after this PR [1] I'll make an update to the blog after that PR is complete


[1] https://github.com/strimzi/strimzi-kafka-operator/pull/7677

Reply:
I think it's also fine to mention the ongoing work to enable it with a link to the PR. A future reader will be able to easily find out the required configuration.

Reply:
It's mentioned at the bottom of the post:
```
In a future version, the Strimzi community hopes to add the option for creating custom Cruise Control API users.
This would remove the requirement for the Cruise Control API security settings to be disabled for applications like the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) to work with Strimzi. Until then, hack away!
```
However, I would like to avoid a link to this PR since it is in early stages